docs/user_guide/mitigation.rst

```diff
@@ -68,6 +68,259 @@ Postprocessing
 
 .. currentmodule:: fairlearn.postprocessing
 
+Fairlearn currently supports one postprocessing technique that is based on the
+paper *Equality of Opportunity in Supervised Learning* [#3]_. Unlike other
+mitigation techniques :class:`ThresholdOptimizer` is built to satisfy the
+specified fairness criteria exactly and with no remaining disparity.
+In many cases this comes at the expense of performance, for example, with
+significantly lower accuracy. Regardless, it is a useful data point to compare
+results with.
+
+:class:`ThresholdOptimizer` expects an estimator that provides it with scores.
+While the output of :class:`ThresholdOptimizer` is binary, the input need not
+be. In fact, real valued input, e.g. from a regressor, provides it with many
+more options to create thresholds. For :math:`n` input data points with
+:math:`m \leq n` different score values it has :math:`m+1` different
+thresholds. At each threshold one can create one of two thresholding rules,
+i.e. functions that indicate which data points get label :code:`1` based on
+their score.
+
+We can think of each thresholding rule as a binary classifier and determine
+its true positive rate and false positive rate.
+The code below visualizes the threshold selection process with an ROC curve.
+The ROC curves consist of the true and false positive rates for each
+of the thresholding rules, with separate ones per sensitive feature value.
+Note that the plot omits points that are within the convex hull of points.
+
+.. plot::
+    :include-source:
+
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from fairlearn.postprocessing import ThresholdOptimizer, plot_threshold_optimizer
+    >>> from sklearn.datasets import fetch_openml
+    >>> import pandas as pd
+    >>> data = fetch_openml(data_id=1590, as_frame=True)
+    >>> X = pd.get_dummies(data.data)
```

Comment:
could we use a `Pipeline` instead of `get_dummies`? See #649

Reply:
Sure, copied that from another place in the repo, but I'm all in favor of moving everything to better practices.

Reply:
Actually, come to think of it, it seems like this code could be reused in many different places. Is there a reason why there isn't a shorter / one-line version of this? Even if it's just done in every example where we use `fetch_adult` it feels like we'd simplify the shown code a lot by putting that code in a utility function.

Reply:
After spending about an hour on this I'd much rather not do this in this PR. This is consistent with other places such as quickstart, and changing all of them goes way beyond the scope of this PR. Nevertheless, I want to address it at some point. This means that we need to be able to pass `Pipeline`s into mitigation techniques and from my experiment that doesn't necessarily work since the feature encoding (e.g. one hot encoding of categorical string features) only happens after our mitigation techniques do some of their own validation (namely `check_X_y`) which results in errors. The cleaner way would probably be to create a `Pipeline` around the mitigation technique, so that the mitigation technique never needs to see the "un-preprocessed" `X`, but I remember that we need to do something more to get that to work. I'd really like to prioritize some of this since it keeps coming up and people expect this to work.

Does this make sense @adrinjalali ? I'll open an issue for this.

Reply:
sure, happy to have it tackled in a different issue.

Reply:
Awesome! --> #665